services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/MapboxV6BaseGeocoding.java

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.api.geocoding.v6;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.mapbox.core.MapboxService;
+
+/**
+ * Base class for entry points to Mapbox V6 geocoding: forward geocoding, reverse geocoding
+ * and batch geocoding. See derived classes for more information.
+ * @param <T> response type.
+ */
+public abstract class MapboxV6BaseGeocoding<T> extends MapboxService<T, V6GeocodingService> {
```

Comment:
Thanks for raising this question, I was going to discuss it because I don't like this option either.

> 3. New major services-geocoding release that would remove all v5 classes (for v5 use prev major release);

v5 is not deprecated and I'm not sure if it's going to be, so it's not an option for now.

> 2. Just the v6 package;

I like this option but I had the same concern as you described.

>  but there are not a lot of conflicting classes, right?

Yes, right. We can leave v6 prefix for entrypoint classes (`V6ForwardGeocodingRequestOptions`, and `MapboxV6BatchGeocoding`), and for few more cases like `V6Context`, `V6ContextElement`. What do you think?

Reply:
> v5 is not deprecated and I'm not sure if it's going to be, so it's not an option for now.

I'm not suggesting to deprecate it, rather use a different major of mapbox-java. However, having 2 active major trains is not a very good idea.

> Yes, right. We can leave v6 prefix for entrypoint classes (V6ForwardGeocodingRequestOptions, and MapboxV6BatchGeocoding), and for few more cases like V6Context, V6ContextElement. What do you think?

Why these specific classes? What's the criteria? IMO that would be: `MapboxV6Geocoding`, `V6GeocodingService`, `V6Context`, `V6ContextElement`, V6Feature`, `V6FeatureType`. These classes have conflicts with v5 and I added ContextElement and FeatureType for consistency (because we have Context and Feature).